lib/engine/step/g_1856/bankrupt.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require_relative '../bankrupt'
+
+module Engine
+  module Step
+    module G1856
+      class Bankrupt < Bankrupt
+        def active?
+          active_entities.any?
+        end
+
+        def active_entities
+          return [] unless @round.cash_crisis_player
+
+          # Rotate players to order starting with the current player
+          players = @game.players.rotate(@game.players.index(@round.cash_crisis_player))
+          [players.find { |p| p.cash.negative? }]
+        end
+
+        def process_bankrupt(action)
+          player = action.entity
+
+          @log << "-- #{player.name} goes bankrupt and sells remaining shares --"
+
+          # next the president sells all normally allowed shares
+          player.shares_by_corporation.each do |corporation, _|
+            next unless corporation.share_price # if a corporation has not parred
+            next unless (bundle = @game.sellable_bundles(player, corporation).max_by(&:price))
+
+            @game.sell_shares_and_change_price(bundle)
+          end
+
+          # finally, move all presidencies into the market, do not change presidency
+          player.shares_by_corporation.each do |corporation, shares|
+            next if shares.empty?
+
+            bundle = ShareBundle.new(shares)
+            @game.sell_shares_and_change_price(bundle, allow_president_change: false)
+
+            next unless corporation.owner == player
+
+            @log << "-- #{corporation.name} enters liquidation (it has no president) --"
+            @game.liquidate!(corporation)
```

Comment:
I don't think I remember 1856 having liquidation

Reply:
That is correct; this is changed in the latest commit - the branch is functionally done so I'm un-drafting it